tests/display.rs

```diff
@@ -1283,6 +1283,131 @@ mod enums {
                 }
             }
         }
+
+        mod shared_format {
+            use super::*;
+            mod single {
+                use super::*;
+
+                #[derive(Display)]
+                #[display("Variant: {_variant}")]
```

Comment:
I think there should be also tests like `#[display("Variant: {}", _variant)]`

Reply:
the `_0` also doesn't work as a positional placeholder. Adding support for positional placeholders is non-trivial, because the current implementation splits the format string and calls `write!` multiple times. So if you have something like:
```rust
#[display("{}, {_variant}, {}", 123, 456)]
```
Then you need to move the expressions to the correct place.
Because this will result in the wrong output:
```rust
write!("{}, ", 123, 456)
// Write variant
write!(", {}", 123, 456)
```

I agree that positional parameters are useful, but I think we only need named parameters for 1.0. In 0.99 we didn't even support parameters at all, only the variant itself. So even with the current implementation we have a superset of the features of 0.99. And that is really all that I think we should be aiming for for 1.0 at this point. Otherwise the release might take another year.

Reply:
@JelteF so, as I see, the problem lays in the way the code desugars.

Why not try to leverage `format_args!()` and do as we usually do?
```rust
let _variant = format_args!("whatever variant formatting is", 1, 2);
write!(f, "{}, {_variant}, {}, {}", 123, 456, _variant)
```

Were there any problems with such design?

Reply:
Ahh... yes... https://github.com/rust-lang/rust/issues/92698  
Let me think...

Reply:
Okay, this desugaring just works:
https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=5868f867756cdddd4585eac6dff1856b

It should also simplify the implementation, as we don't need all the machinery to cut the expression to separate `write!`s.

Reply:
If you're short on time with this now, I may pick up and re-implement, if you don't mind.

Reply:
That desugaring is a lot better for sure. So yes, if you have time now, please re implement this.